example-model/src/main/java/uk/gov/gchq/palisade/example/client/SimpleClient.java

```diff
@@ -83,16 +82,17 @@ private DataRequestResponse makeRequest(final String fileName, final String reso
         requireNonNull(response, "response");
 
         final List<Stream<T>> dataStreams = new ArrayList<>(response.getResources().size());
-        for (final Entry<LeafResource, ConnectionDetail> entry : response.getResources().entrySet()) {
-            final ConnectionDetail connectionDetail = entry.getValue();
+        for (final LeafResource resource : response.getResources()) {
+            final ConnectionDetail connectionDetail = resource.getConnectionDetail();
             final URI dataService = new URI(connectionDetail.createConnection());
             final RequestId uuid = response.getOriginalRequestId();
 
             final ReadRequest readRequest = new ReadRequest()
                     .token(response.getToken())
-                    .resource(entry.getKey());
+                    .resource(resource);
             readRequest.setOriginalRequestId(uuid);
 
+            LOGGER.info("Resource {} has DATA-SERVICE connection detail {}", resource.getId(), connectionDetail);
             InputStream responseStream = dataClient.readChunked(dataService, readRequest).body().asInputStream();
```

Comment:
The dataService value should be updated accordingly
```suggestion
            InputStream responseStream = dataClient.readChunked(dataService.getUri(), readRequest).body().asInputStream();
```

Reply:
Won't do, see above and Palisade-services